python_modules/dagster/dagster/core/execution.py

```diff
@@ -701,6 +706,41 @@ def invoke_executor_on_plan(pipeline_context, execution_plan, step_keys_to_execu
         yield step_event
 
 
+def _check_reexecution_config(pipeline_context, execution_plan, run_config):
+    check.invariant(pipeline_context.run_storage)
+
+    previous_run_id = run_config.reexecution_config.previous_run_id
+
+    if not pipeline_context.run_storage.has_run(previous_run_id):
+        raise DagsterRunNotFoundError(
+            'Run id {} set as previous run id was not found run storage'.format(previous_run_id),
+            invalid_run_id=previous_run_id,
+        )
+
+    for step_output_handle in run_config.reexecution_config.step_output_handles:
+        if not execution_plan.has_step(step_output_handle.step_key):
+            raise DagsterExecutionStepNotFoundError(
+                (
+                    'Step {step_key} was specified as a step from a previous run. '
```

Comment:
couple oddities.

- what if the pipeline has changed between runs? (the step existed in the last run, but not in this one)
- what if a step with the same name existed in the previous run, but the previous run was on a different pipeline?

Reply:
yup. i believe we are going to have to deal with client side/api consumer side. e.g. they will have to know about rename changes and configure step output handles and step keys appropriately